File: c++/classes/class_static_variable_access.cpp
#include <iostream>
#include <string>

class Token {
public:
  static int number;
  std::string color;
};

int Token::number;

int main() {
  Token::number = 4;

  std::cout << "Token::number: " << Token::number << std::endl;

  Token some_token;

  some_token.number = 5;
  some_token.color = "magenta";

  std::cout << "some_token.number: " << some_token.number << ", some_token.color: " << some_token.color << std::endl;

  Token *pointer_to_token = &some_token;

  (*pointer_to_token).number = 6;
  (*pointer_to_token).color = "amaranthine";

  std::cout << "(*pointer_to_token).number: " << (*pointer_to_token).number << ", (*pointer_to_token).color: " << (*pointer_to_token).color << std::endl;

  pointer_to_token->number = 7;
  pointer_to_token->color = "indigo";

  std::cout << "pointer_to_token->number: " << pointer_to_token->number << ", pointer_to_token->color: " << pointer_to_token->color << std::endl;

  Token &reference_to_token = some_token;

  reference_to_token.number = 8;
  reference_to_token.color = "navy";

  std::cout << "reference_to_token.number: " << reference_to_token.number << ", reference_to_token.color: " << reference_to_token.color << std::endl;
}

File: python/classes/class_static_variable_access.py
# Copyright (C) 2019-2026 Katarzyna Krasińska
# Languagium - https://github.com/katheroine/languagium
# Licensed under GPL-3.0 - see LICENSE.md
# SPDX-License-Identifier: GPL-3.0-or-later

class Token:
    number = 0

    def __init__(self):
        Token.number = 0
        self.color = ""

Token.number = 4

print(f"Token.number: {Token.number}")

some_token = Token()

some_token.number = 5
some_token.color = "magenta"

print(f"some_token.number: {some_token.number}, some_token.color: {some_token.color}");
